src/timetable_cmd.cpp

```diff
@@ -222,8 +222,23 @@ CommandCost CmdSetVehicleOnTime(DoCommandFlag flags, VehicleID veh)
 	if (ret.Failed()) return ret;
 
 	if (flags & DC_EXEC) {
-		v->lateness_counter = 0;
-		SetWindowDirty(WC_VEHICLE_TIMETABLE, v->index);
+        if (apply_to_group){
+            int32 most_late = 0;
+            for (Vehicle *u = v->FirstShared(); u != nullptr; u = u->NextShared()) {
+                if (u->lateness_counter > most_late){
+                    most_late = u->lateness_counter;
+                }
+            }
+            if (most_late > 0) {
+                for (Vehicle *u = v->FirstShared(); u != nullptr; u = u->NextShared()) {
+                    u->lateness_counter -= most_late;
+                    SetWindowDirty(WC_VEHICLE_TIMETABLE, u->index);
+                }
+            }
+        } else{
+            v->lateness_counter = 0;
+            SetWindowDirty(WC_VEHICLE_TIMETABLE, v->index);
+        }
```

Comment:
This all needs one more indentation level, since it's within the `if (flags & DC_EXEC)` block.

Reply:
Can you please elaborate where exactly to change indentation? The contents of `if (flags & DC_EXEC)` seem indented to me.